internal/controller/api_to_config.go

```diff
@@ -684,6 +688,20 @@ func validatePrefixes(prefixes []string) error {
 	return nil
 }
 
+func validateNeighbors(a []v1beta1.Neighbor) error {
```

Comment:
this seems to cover only the specific case of mismatch between address and asns, while iirc there are more fields that can cause a similar conflict. would validating by merging the router with itself achieve the same goal?

Reply:
Refactored to re-use the merging logic